docs/content/preview/api/ysql/the-sql-language/statements/ddl_create_table.md

```diff
@@ -115,6 +118,17 @@ This relaxation primarily aims to provide increased flexibility for carrying out
 
 Note that the [INSERT](../dml_insert/) command can only accommodate one override clause for an entire statement. As a result, having several identity columns, each exhibiting distinct behaviours, is not effectively supported.
 
+### Stored generated columns
+
+A stored generated column is computed when it is written (inserted or updated) and occupies storage as if it were a normal column. A generated column cannot be written to directly. In INSERT or UPDATE commands, a value cannot be specified for a generated column, but the keyword DEFAULT may be specified. This feature is particularly useful for scenarios requiring precomputed values for indexing, sorting, or filtering, as it reduces computation overhead during queries.
+
+Several restrictions apply to the definition of generated columns and tables involving generated columns:
+
+- The generation expression can only use immutable functions and cannot use subqueries or reference anything other than the current row in any way.
+- A generation expression cannot reference another generated column.
+- A generation expression cannot reference a system column, except tableoid.
+- A generated column cannot have a column default or an identity definition.
+- A generated column cannot be part of a partition key.
```

Comment:
We can enhance this to add restrictions on non-partition key columns:

* For partitioned tables:
    * A generated column cannot be part of a partition key.
    * If a parent column is a generated column, a child column must also be a generated column using the same expression.
    * If a parent column is not a generated column, a child column may be defined to be a generated column or not.

Reply:
is the following part deliberately skipped?

Additional considerations apply to the use of generated columns.

Generated columns maintain access privileges separately from their underlying base columns. So, it is possible to arrange it so that a particular role can read from a generated column but not from the underlying base columns.

Generated columns are, conceptually, updated after BEFORE triggers have run. Therefore, changes made to base columns in a BEFORE trigger will be reflected in generated columns. But conversely, it is not allowed to access generated columns in BEFORE triggers.

Generated columns are skipped for logical replication and cannot be specified in a CREATE PUBLICATION column list.